lib/matterhorn/links/relation/base.rb

```diff
@@ -124,9 +124,11 @@ def find_with_ids(resource, ids)
           scope(resource).in(inverse_field_key => ids)
         end
 
-        def get_items_ids(items)
+        def get_items_ids(resource, items)
+          resource = [resource].flatten.inject({}){ |hsh, object| hsh[object[:_id]] = object; hsh}
```

Comment:
I have moved the generation of the resource hash to be outside of the link generation so there is no need to memoize it now, how would you test it now that it's passed down?